### PR TITLE
Add scanning for corrupt ActivityIds

### PR DESCRIPTION
**Pull request by Ardagan**
**What changed?**
Added validation of activity ids to tctl

**Why?**
Debugging corrupt activity ids.

**How did you test it?**
local execution

**Potential risks**

### CODE REVIEW

File: tools/cli/adminDBScanCommand.go [hunk 1]
```diff
@@ -395,6 +398,26 @@ func scanShard(
...
+			switch activityIdsVerificationResult {
+			case VerificationResultNoCorruption:
+			case VerificationResultDetectedCorruption:
+				report.Scanned.CorruptionTypeBreakdown.TotalActivityIdsCorrupted++
+				report.Scanned.CorruptedExecutionsCount++
+				continue
+			case VerificationResultCheckFailure:
+				report.Scanned.ExecutionCheckFailureCount++
+				continue
+			}
```
Comment (wxing1292) [inline tools/cli/adminDBScanCommand.go:420]: can we refactor this giant for loop into several dedicated functions?

File: tools/cli/adminDBScanCommand.go [hunk 2]
```diff
@@ -564,6 +587,48 @@ func verifyFirstHistoryEvent(
...
+	nextEventID int64,
+	activityInfos map[int64]*persistencespb.ActivityInfo,
+	executionInfo *persistencespb.WorkflowExecutionInfo,
+	executionState *persistencespb.WorkflowExecutionState,
+	corruptedExecutionWriter BufferedWriter,
+	branch *persistencespb.HistoryBranch,
+	) (
+	VerificationResult,
+) {
+	if (len(activityInfos) > 0) {
```
Comment (wxing1292) [inline tools/cli/adminDBScanCommand.go:603]:
```go
if len(activityInfos) == 0 {
    return VerificationResultNoCorruption
}

for activityId, _ := range activityInfos {
    ...
}
```

File: tools/cli/adminDBScanCommand.go [hunk 3]
```diff
@@ -564,6 +587,48 @@ func verifyFirstHistoryEvent(
...
+		for activityId, _ := range activityInfos {
+			if activityId >= nextEventID {
```
Comment (wxing1292) [inline tools/cli/adminDBScanCommand.go:605]:
```go
if activityId >= nextEventID || activityId <= 0 {
```

File: tools/cli/adminDBScanCommand.go [hunk 4]
```diff
@@ -564,6 +587,48 @@ func verifyFirstHistoryEvent(
...
+				byteBranch, err := byteKeyFromProto(branch)
+				if err != nil {
+					return VerificationResultCheckFailure
+				}
```
Comment (wxing1292) [inline tools/cli/adminDBScanCommand.go:609]: lazy initialization?

Comment (Ardagan) [reply tools/cli/adminDBScanCommand.go:609]: Can you elaborate please?

Comment (wxing1292) [reply tools/cli/adminDBScanCommand.go:609]: `byteBranch, err := byteKeyFromProto(branch)` this function is invoked per `activityInfo` within `activityInfos`
the branch object does not change, so either do `byteBranch, err := byteKeyFromProto(branch)` at the very beginning or do lazy initialization

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**